app/src/main/java/org/mozilla/fenix/RunBlockingCounter.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This class implements the alternative ways to invoke runBlocking with some
+// monitoring by wrapping the raw methods. This lint check tells us not to use the raw
+// methods so we suppress the check.
+@file:Suppress("MozillaRunblockingCheck")
+
+package org.mozilla.fenix
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.runBlocking
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * Counts the number of runBlocking calls made
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+class RunBlockingCounter {
+    companion object {
+        var runBlockingCount = 0L
```

Comment:
Given this runs in production code, I'm starting to think we should prevent overflow. The app shouldn't have a hard limit on how long it can run. Maybe in the setter? We could probably change it back to an Int in that case.

Reply:
I can always add it to the `runBlockingIncrement` function. I just felt uneasy about adding more functionality (albeit its probably minimal) to a function that's just suppose to wrap one provided by Kotlin

Reply:
Why do you want to avoid adding too much functionality to a wrapping method?